components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/AuthorizedAPIManagementServiceImpl.java

```diff
@@ -181,4 +189,38 @@ private IdentityApplicationManagementServerException buildServerException(String
 
         return new IdentityApplicationManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(), message, ex);
     }
+
+    private void updateRolesWithRemovedScopes(String appId, List<String> removedScopes, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (CollectionUtils.isEmpty(removedScopes) || !isApplicationAudience(appId, tenantDomain)) {
```

Comment:
can't we use `removedScopes.isEmpty()` here?

Reply:
CollectionUtils.isEmpty will handle null case too